src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -118,5 +126,103 @@ private void EnsureMvcInitialized()
+
+                var fakeHelperType = typeof(ViewComponentTagHelperDesignTimeHelperType).FullName;
+                var fakeHelperDescriptor = new TagHelperDescriptor(descriptor);
+                fakeHelperDescriptor.TypeName = fakeHelperType;
+
+                var fakeHelperAttributes = new List<TagHelperAttributeDescriptor>();
+                fakeHelperDescriptor.Attributes = fakeHelperAttributes;
+
+                foreach (var attribute in descriptor.Attributes)
+                {
+                    var summary = $"{attribute.TypeName}: {descriptor.TypeName}.{attribute.PropertyName}";
+                    var fakeVariableName = $"__obj";
+                    var fakeAttributePropertyName = $"{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderProperty)} = null; " +
+                            $"{fakeType}.{nameof(ViewComponentTagHelperDesignTimeType.ActionProperty)} = () => {{ " +
+                            $"{attribute.TypeName} {fakeVariableName} = default({attribute.TypeName}); " +
+                            $"{fakeType}.{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderMethod)}({fakeVariableName}); {fakeVariableName} ";
+
+                    var fakeAttribute = new TagHelperAttributeDescriptor()
+                    {
+                        IsIndexer = attribute.IsIndexer,
+                        IsEnum = attribute.IsEnum,
+                        IsStringProperty = attribute.IsStringProperty,
+                        Name = attribute.Name,
+                        PropertyName = fakeAttributePropertyName,
+                        TypeName = attribute.TypeName,
+                        DesignTimeDescriptor = new TagHelperAttributeDesignTimeDescriptor
+                        {
+                            Summary = attribute.DesignTimeDescriptor?.Summary ?? summary
+                        },
+                    };
+                    fakeAttributes.Add(fakeAttribute);
+
+                    var fakeHelperAttributePropertyName =
+                        $"{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderProperty)} = null; }};//";
+
+                    var fakeHelperAttribute = new TagHelperAttributeDescriptor()
+                    {
+                        IsIndexer = attribute.IsIndexer,
+                        IsEnum = attribute.IsEnum,
+                        IsStringProperty = attribute.IsStringProperty,
+                        Name = attribute.Name,
+                        PropertyName = fakeHelperAttributePropertyName,
+                        TypeName = attribute.TypeName,
+                    };
+                    fakeHelperAttributes.Add(fakeHelperAttribute);
+                }
+
+                fakeDescriptors.Add(fakeDescriptor);
+                fakeDescriptors.Add(fakeHelperDescriptor);
+            }
+
+            return fakeDescriptors;
+        }
+    }
+
+    public class ViewComponentTagHelperDesignTimeType : ITagHelper
```

Comment:
- Inherit from `TagHelper` so you don't have to specify everything.
- Mark this and the class below as `abstract` so if a user tries to `@addTagHelper` this assembly [this](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperConventions.cs#L21) returns false.